packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -729,7 +729,7 @@ export class StringListParameter extends ParameterBase implements IStringListPar
       name: this.physicalName,
       tier: props.tier,
       type: ParameterType.STRING_LIST,
-      value: props.stringListValue.join(','),
+      value: Fn.join(',', props.stringListValue),
```

Comment:
Necessary when the value is a token; this isn't a breaking change, right?

Reply:
Honestly couldn't tell you 😅.

Not if `Fn.join` short-circuits literals, but yes if it doesn't. Is there a test for this?

Reply:
I don't think so, but what does it mean to short-circuit a literal?

Reply:
>  Is there a test for this?

☞ meant as ☞

"Please add a test for this"

🤣

---------

> what does it mean to short-circuit a literal?

`Fn.join(',', ['a', 'b'])` could be rendered to either:

* `{ "Fn::Join": [",", ["a", "b"]] }`; or
* `"a,b"`

Both have the same value when finally passed to the resource -- but we can only do the second one if all input values are *concrete*.

I guess in that sense the change is never **breaking**, but it may be distressing to users if their template contents change all of a sudden. We try to minimize that happening needlessly. Although I could be convinced that in this case the change really is safe.

Reply:
It's the second one, and in fact we have these tests from core:

```
    test('pre-concatenates string literals', asyncTest(async () => {
      const stack = new Stack();
      fc.assert(
        fc.property(
          fc.string(), fc.array(nonEmptyString, 1, 15),
          (delimiter, values) => stack.resolve(Fn.join(delimiter, values)) === values.join(delimiter),
        ),
        { verbose: true },
      );
    }));

    test('pre-concatenates around tokens', asyncTest(async () => {
      const stack = new Stack();
      fc.assert(
        fc.property(
          fc.string(), fc.array(nonEmptyString, 1, 3), tokenish, fc.array(nonEmptyString, 1, 3),
          (delimiter, prefix, obj, suffix) =>
            _.isEqual(stack.resolve(Fn.join(delimiter, [...prefix, stringToken(obj), ...suffix])),
              { 'Fn::Join': [delimiter, [prefix.join(delimiter), obj, suffix.join(delimiter)]] }),
        ),
        { verbose: true, seed: 1539874645005, path: '0:0:0:0:0:0:0:0:0' },
      );
    }));
```